File: pyg_manager_web/src/main/java/com/pyg/manager/controller/BrandController.java
package com.pyg.manager.controller;

import bean.PageResult;
import bean.Result;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.pojo.TbBrand;
import com.pyg.sellergoods.service.BrandService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference  //远程注入
    private BrandService brandService;
    /**
     * 此方法是在添加模板时需要的 要求数据格式是 [{"id":"1","text":"华为"},{"id":"2","text":"小米"},{"id":"3","text":"锤子"}]
     */
    @RequestMapping("/findBrandList")
    public List<Map> findBrandList(){
        return brandService.findBrandList();
    }


    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }


    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNo, Integer pageSize){
//        {total:100,rows:[{},{}]}
        return brandService.findPage(pageNo,pageSize);
//        return null;
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){ //@RequestBody是用来接收json数据的
//        {success:true|false,message:"添加成功"|"添加失败"}
        try {
            brandService.add(brand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
//        return null;
    }
    @RequestMapping("/fineOne")
    public TbBrand fineOne(Long id){
          return   brandService.findOne(id);
//        return null;
    }
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){ //@RequestBody是用来接收json数据的
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    @RequestMapping("/dele")
    public Result dele(Long[] ids){ //@RequestBody是用来接收json数据的
        try {
            brandService.dele(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(Integer pageNo, Integer pageSize,@RequestBody TbBrand tbBrand){
//        {total:100,rows:[{},{}]}
        return brandService.findPage(tbBrand,pageNo,pageSize);
//        return null;
    }
}

File: pyg_manager_web/src/main/webapp/js/controller/brandController.js
// 表现层  和页面数据交互
app.controller("brandController",function ($scope,brandService,$controller) {


    $controller("baseController",{$scope:$scope}); //brandController继承baseController的语法 extends

    $scope.findAll=function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        })
    }


    // 保存方法
    $scope.save=function () {
        var obj=null;
        if($scope.entity.id!=null){
            obj = brandService.update($scope.entity);
        }else{
           obj = brandService.add($scope.entity);
        }
        obj.success(function (response) {
            // response ={success:true|false,message:"添加成功"|"添加失败"}
            if(response.success){
                // 列表的数据刷新
                $scope.reloadList();
            }else{
                alert(response.message);
            }
        })
    }

    // 根据id查询对象
    $scope.findOne=function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity=response;
        })
    }


    // 删除
    $scope.dele=function () {
        // 判断数组中是否有值
        if($scope.selectIds.length==0){
            alert("请选择您需要删除的数据!");
            return;
        }
        var flag = window.confirm("确定要删除您选择的数据吗?");
        if(flag){
           brandService.dele($scope.selectIds).success(function (response) {
                if(response.success){
                    $scope.reloadList();//刷新列表数据
                    $scope.selectIds=[];// 清空数组
                }else{
                    alert(response.message);
                }
            })
        }
    }

    $scope.searchEntity={};

    $scope.search=function (pageNo,pageSize) {
        // $scope.searchEntity   页码  条数
        brandService.search(pageNo,pageSize,$scope.searchEntity).success(function (response) {
            // response:分页后的结果  当前也得数据list 总条数  {total:100,rows:[{},{}]}
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    }



})